### PR TITLE
Update to Jan 2021 3.1.11 release.

### PR DESCRIPTION
**Pull request by crummel**

### CODE REVIEW

**Review by dseefeld — APPROVED** [review]

File: tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/RemoveInternetSourcesFromNuGetConfig.cs [hunk 1]
```diff
@@ -67,6 +68,9 @@ public override bool Execute()
 
             packageSourcesElement.ReplaceNodes(local.ToArray());
 
+            // Remove disabledPackageSources element so if any internal packages remain, they are used in source-build
+            disabledPackageSourcesElement?.ReplaceNodes(new XElement("clear"));
```
Comment (dagood) [inline tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/RemoveInternetSourcesFromNuGetConfig.cs:72]: Hmm... do you know why this works for non-internal builds? I would expect it to try to use internal sources in the production build and fail, if we remove the disables unconditionally. It obviously didn't fail, but curious why.

**Review by dagood — APPROVED** [review]

File: tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/RemoveInternetSourcesFromNuGetConfig.cs (hunk 1 above)
Comment (crummel) [reply tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/RemoveInternetSourcesFromNuGetConfig.cs:72]: I don't.  My guess would be it's hitting the public feeds first, we've seen some nondeterministic behavior with which NuGet feed gets hit first before.  Think it's worth removing this?

Comment (dagood) [reply tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/RemoveInternetSourcesFromNuGetConfig.cs:72]: I think we should go ahead with and merge the PR, but also file an issue to look into it.

Comment (dagood) [reply tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/RemoveInternetSourcesFromNuGetConfig.cs:72]: Ahh right, poked around a bit, and we actually remove all internal sources all the time unless specifically included by specific projects when they need an internal prebuilt dependency like templates:

https://github.com/dotnet/source-build/blob/8cad9c79cb726997804c415782eaac07168e40d8/repos/core-sdk.proj#L77-L84

That only happens when auth is provided, so it works fine externally. Case closed. 😄

Comment (crummel) [reply tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/RemoveInternetSourcesFromNuGetConfig.cs:72]: Cool, thanks for looking into it!

**Comment by crummel** [comment]
OSX failures are known issues: https://github.com/dotnet/source-build/issues/1977, https://github.com/dotnet/source-build/issues/1794.

**Pull request merged**